libretro-common/include/libretro.h

```diff
@@ -976,6 +976,19 @@ struct retro_hw_render_context_negotiation_interface
                                             * so it will be used after SET_HW_RENDER, but before the context_reset callback.
                                             */
 
+struct retro_resource
+{
+   const char *rel_path;
+   const char *base_path;
```

Comment:
Well, if the frontend path is longer than the size_t for some reason, the core needs to know how big the buffer should be so it can allocate a proper one. Something like:

```
struct retro_resource_path *data = ...;
data->size = strlcpy(data->buffer, resolved_path, data->size);
```